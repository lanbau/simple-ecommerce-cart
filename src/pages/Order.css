@media only screen and (max-width: 600px) {
    .order {
        flex-direction: column;
    }
}

.orders {
    width: 100%;
    padding: 50px;
    display: flex;
    align-items: center;
    /* text-align: center; */
    flex-direction: column;
    margin: 0 16px;
}

.order-title {
    font-size: 32px;
    margin-bottom: 16px;
}

.order {
    padding: 50px;
    margin-bottom: 8px;
    border: 1px solid rgba(201, 201, 201, 0.336);
}