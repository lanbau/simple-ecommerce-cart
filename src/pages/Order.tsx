import React from 'react'
import { useDispatch, useSelector } from 'react-redux'
import './Order.css';
import getOrders from '../selectors/orderSelector';
import convertDate from '../utils/convertDate'
// TODO moment js doesn't work for Jest
const Order = () => {
    const orders = useSelector(getOrders)

    return <div style={{ display: 'flex', flexDirection: 'column', justifyContent: 'center', alignItems: 'center', marginTop: 16 }}>
       { !orders.length && 'No Orders Made.'}
        { orders.map((order, index) => <div className="orders" key={order.id}>
            <div className="order-title">Order Placed: {convertDate(order.date)} </div>
            <div>
                
                { order?.items?.length > 0 && order?.items?.map(({ userId, src, title, body, price }) => {
                        return <div className="order" style={{ display: 'flex' }} key={userId}>
                            <img src={ src } style={{ width: 150, height: 150, marginRight: 16, marginBottom: 8, borderRadius: 10  }}/>
                            <div>
                                <div style={{ fontSize: 24, fontWeight: 'bold' }}>{title}</div>
                                <div style={{ maxWidth: 400 }}>{`${body}`}</div>
                                <div style={{ fontWeight: 'bold' }}>Price: ${price}</div>
                            </div>
                        </div>
                    })
                }
            </div>
            
        </div>)

       }
    </div>
}
export default Order