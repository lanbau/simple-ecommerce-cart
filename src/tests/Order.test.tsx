import React from 'react'
import { useSelector } from 'react-redux'
import Order from '../pages/Order'
import renderer from 'react-test-renderer'

jest.mock('react-redux', () => ({
  useDispatch: jest.fn(),
  useSelector: jest.fn(),
}))

describe('Orders Page', () => {
  test('renders orders', () => {
    useSelector.mockImplementation((selector) => selector({
      orders: [
        {
          id: '123',
          date: '2020-08-26T15:16:25.542Z',
          items: [
            {
              userId: 1,
              id: 2,
              title: 'qui est esse',
              body: 'est rerum tempore vitae\nsequi sint nihil reprehenderit dolor beatae ea dolores neque\nfugiat blanditiis voluptate porro vel nihil molestiae ut reiciendis\nqui aperiam non debitis possimus qui neque nisi nulla',
              price: 41,
              src: 'https://picsum.photos/200/300?random=1'
            },
            {
              userId: 2,
              id: 2,
              title: 'qui est esse',
              body: 'est rerum tempore vitae\nsequi sint nihil reprehenderit dolor beatae ea dolores neque\nfugiat blanditiis voluptate porro vel nihil molestiae ut reiciendis\nqui aperiam non debitis possimus qui neque nisi nulla',
              price: 41,
              src: 'https://picsum.photos/200/300?random=1'
            }
          ]
        }
      ]
    }))
    const container = renderer
      .create(
        <Order />
      )
      .toJSON()
    expect(container).toMatchSnapshot()
  })
})