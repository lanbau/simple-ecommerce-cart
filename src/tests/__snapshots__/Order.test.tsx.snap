// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Orders Page renders orders 1`] = `
<div
  style={
    Object {
      "alignItems": "center",
      "display": "flex",
      "flexDirection": "column",
      "justifyContent": "center",
      "marginTop": 16,
    }
  }
>
  <div
    className="orders"
  >
    <div
      className="order-title"
    >
      Order Placed: 
      2020-08-26
       
    </div>
    <div>
      <div
        className="order"
        style={
          Object {
            "display": "flex",
          }
        }
      >
        <img
          src="https://picsum.photos/200/300?random=1"
          style={
            Object {
              "borderRadius": 10,
              "height": 150,
              "marginBottom": 8,
              "marginRight": 16,
              "width": 150,
            }
          }
        />
        <div>
          <div
            style={
              Object {
                "fontSize": 24,
                "fontWeight": "bold",
              }
            }
          >
            qui est esse
          </div>
          <div
            style={
              Object {
                "maxWidth": 400,
              }
            }
          >
            est rerum tempore vitae
sequi sint nihil reprehenderit dolor beatae ea dolores neque
fugiat blanditiis voluptate porro vel nihil molestiae ut reiciendis
qui aperiam non debitis possimus qui neque nisi nulla
          </div>
          <div
            style={
              Object {
                "fontWeight": "bold",
              }
            }
          >
            Price: $
            41
          </div>
        </div>
      </div>
      <div
        className="order"
        style={
          Object {
            "display": "flex",
          }
        }
      >
        <img
          src="https://picsum.photos/200/300?random=1"
          style={
            Object {
              "borderRadius": 10,
              "height": 150,
              "marginBottom": 8,
              "marginRight": 16,
              "width": 150,
            }
          }
        />
        <div>
          <div
            style={
              Object {
                "fontSize": 24,
                "fontWeight": "bold",
              }
            }
          >
            qui est esse
          </div>
          <div
            style={
              Object {
                "maxWidth": 400,
              }
            }
          >
            est rerum tempore vitae
sequi sint nihil reprehenderit dolor beatae ea dolores neque
fugiat blanditiis voluptate porro vel nihil molestiae ut reiciendis
qui aperiam non debitis possimus qui neque nisi nulla
          </div>
          <div
            style={
              Object {
                "fontWeight": "bold",
              }
            }
          >
            Price: $
            41
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
